pennylane/ops/qubit.py

```diff
@@ -364,6 +364,49 @@ def _eigvals(cls, *params):
         return cls.eigvals
 
 
+class CY(Operation):
+    r"""CY(wires)
+    The controlled-Y operator
+
+    .. math:: CY = \begin{bmatrix}
+            1 & 0 & 0 & 0 \\
+            0 & 1 & 0 & 0\\
+            0 & 0 & 0 & -i\\
+            0 & 0 & i & 0
+        \end{bmatrix}.
+
+    .. note:: The first wire provided corresponds to the **control qubit**.
+
+    **Details:**
+
+    * Number of wires: 2
+    * Number of parameters: 0
+
+    Args:
+        wires (Sequence[int] or int): the wires the operation acts on
+    """
+    num_params = 0
+    num_wires = 2
+    par_domain = None
+    matrix = np.array(
+        [
+            [1, 0, 0, 0],
+            [0, 1, 0, 0],
+            [0, 0, 0, -1j],
+            [0, 0, 1j, 0],
+        ]
+    )
+
+    @classmethod
+    def _matrix(cls, *params):
+        return cls.matrix
```

Comment:
Is this the "main implementation" of a gate? In other words, is this method the one that is used for all calculations when a particular gate is in circuits?

Reply:
It's up to the plugin. Some plugins may choose to read the matrix property, and use this to implement the gate (e.g., `default.qubit`). Others may have their own implementation, that doesn't require accessing this property.